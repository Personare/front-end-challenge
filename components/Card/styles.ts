import styled, {css} from 'styled-components'

export const Card = styled.div`
  margin: .6rem;
`
export const CardContent = styled.div<{flipped:boolean}>`
  ${({flipped}) => css`
    position: relative;
    width: 120px;
    height: 200px;
    transition: transform 606ms ease-in-out;
    transform-style: preserve-3d;

    ${flipped && css`
      transform: rotateY(-180deg);
    `}
  `}  
`
const CardFace = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  backface-visibility: hidden;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 3rem;
  cursor: pointer;
  border-radius: 5px;
`
export const CardFaceFront = styled(CardFace)`
  width: 120px;
  height: 200px;
  overflow: hidden;
`
export const CardFaceBack = styled(CardFace)`
  align-self: center;
  background: #420027b9;
  border: 5px solid #ccc;   
  background-position: 50%;
  background-repeat: no-repeat;
  background-image: url('backCard.svg');
  transform: rotateY(-180deg);
`
