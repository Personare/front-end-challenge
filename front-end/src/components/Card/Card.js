import React from "react";
import classNames from "classnames";

import "./card.scss";

const Card = ({ imagesUrl, imageBackCard, card, onClick }) => (
    <div
        role="presentation"
        className={classNames("card", { hover: card.isActive })}
        onClick={onClick.bind(this, card)}
    >
        <div className="card__flipper">
            <div
                className="card__page card__page--front"
                style={{ background: `url(${imageBackCard})` }}
            />
            <div
                className="card__page card__page--back"
                style={{ background: `url(${imagesUrl}${card.image})` }}
            >
                <div className="card__name">{card.name}</div>
            </div>
        </div>
    </div>
);

export default Card;
