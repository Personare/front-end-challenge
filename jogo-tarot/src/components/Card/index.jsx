import { CardContainer, CardDescription, CardFace, Scene } from "./style";
import React, { useContext } from "react";
import { CardContext } from "../../contexts/cardContext";
import { Container } from "../Container/style";

const Card = () => {
    const { card, image_path, back_card, flip } = useContext(CardContext);

    return (
        <Container>
            {card.map((card) => {
                return (
                    <Scene>
                        <CardContainer
                            className={`card ${flip ? "is-flipped" : ""}`}
                        >
                            <CardFace className="card__face--front">
                                <img
                                    src={`${image_path}${card.image}`}
                                    alt={card.name}
                                />
                                <CardDescription>
                                    Lorem ipsum dolor sit amet consectetur
                                    adipisicing elit.
                                </CardDescription>
                            </CardFace>
                            <CardFace className="card__face--back">
                                <img src={`${back_card}`} alt={card.name} />
                            </CardFace>
                        </CardContainer>
                    </Scene>
                );
            })}
        </Container>
    );
};

export default Card;
