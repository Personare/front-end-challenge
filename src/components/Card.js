import React, { PropTypes } from 'react';
import TweenLite from 'gsap';

export class Card extends React.Component {
    componentWillReceiveProps(nextProps) {
        if (nextProps.selected !== this.props.selected) {
            TweenLite.to(this.refs.cardFaces, 1, { rotationY: !nextProps.selected ? -180 : 0 });
        }
    }

    render() {
        return (
            <article className="card" onClick={this.props.onClick}>
                <div className="card-faces" ref="cardFaces">
                    <div className="front">
                        <img src={this.props.frontImg} alt={this.props.name} />
                    </div>
                    <div className="back">
                        <img src={this.props.backImg} alt={this.props.name} />
                    </div>
                </div>
            </article>
        );
    }
}

Card.propTypes = {
    name: PropTypes.string,
    frontImg: PropTypes.string,
    backImg: PropTypes.string,
    selected: PropTypes.bool,
    onClick: PropTypes.func,
};
