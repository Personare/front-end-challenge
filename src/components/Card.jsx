import React from "react";

const Card = ({ name = "CartjÃªnha", image = "", ...rest }) => {
  return (
    <div className="card">
      <img src={image} alt={name} />
    </div>
  );
};

export default Card;
