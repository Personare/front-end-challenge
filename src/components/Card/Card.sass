.CardContainer
  position: relative
  height: 210px
  cursor: pointer
  margin: 10px;
  .CardContent
    transform-style: preserve-3d

  .CardImage, .BackImage
    position: absolute
    width: 100px
    height: 210px
    overflow: hidden
    border-radius: 5px
    backface-visibility: hidden
    > img
      width: 100px
      height: 210px

  .BackImage
    transform: rotateY(-180deg)
