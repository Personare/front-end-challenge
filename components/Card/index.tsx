import Image from 'next/image'
import * as S from './styles'

type CardProps = {
  imagesUrl: string;
  name: string;
  flipped?: boolean;
  image: string;
  handleClick?:(name: string)=> void;
}
const Card = ( {flipped = false, image, handleClick, name, imagesUrl}: CardProps) => {
  const handleClickfn = (name: string) =>{
    if(handleClick){
      handleClick(name)
    }
  }
  return(
    <S.Card onClick={() => handleClickfn(name)}>           
      <S.CardContent flipped={flipped}>
        <S.CardFaceBack />
        <S.CardFaceFront>
          <Image width={120} height={200} src={`${imagesUrl}${image}`} alt={name} />
        </S.CardFaceFront>
      </S.CardContent>
    </S.Card> 
  )
}
export default Card;
