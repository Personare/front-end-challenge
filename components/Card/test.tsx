import { render } from "@testing-library/react"
import Card from './'

describe('components/Card', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Card image='test.jpg' name='test' imagesUrl='http://www.com.br' />)
    expect(container).toMatchSnapshot()
  })
  
})
