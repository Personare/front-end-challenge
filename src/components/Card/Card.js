import React, { Component } from 'react';
import TweenMax from 'gsap';

import './Card.sass';

export class Card extends Component {
  constructor(props) {
    super(props);
    this.cardContentRef = React.createRef();
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.selected !== this.props.selected) {
      TweenMax.to(this.cardContentRef.current, 1, { rotationY: !nextProps.selected ? -180 : 0 });
    }
  }

  render() {
    return (
      <article className="CardContainer" onClick={this.props.click}>
        <div className="CardContent" ref={this.cardContentRef}>
          <div className="CardImage">
            <img src={this.props.card} alt={this.props.name} />
          </div>
          <div className="BackImage">
            <img src={this.props.imageBack} alt={this.props.name} />
          </div>
        </div>
      </article>
    );
  }
}

export default Card;
