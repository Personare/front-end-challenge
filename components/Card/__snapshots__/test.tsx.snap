// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Card Should match snapshot 1`] = `
.c0 {
  margin: .6rem;
}

.c1 {
  position: relative;
  width: 120px;
  height: 200px;
  -webkit-transition: -webkit-transform 606ms ease-in-out;
  -webkit-transition: transform 606ms ease-in-out;
  transition: transform 606ms ease-in-out;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-transform: rotateY(-180deg);
  -ms-transform: rotateY(-180deg);
  transform: rotateY(-180deg);
}

.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 3rem;
  cursor: pointer;
  border-radius: 5px;
}

.c4 {
  width: 120px;
  height: 200px;
  overflow: hidden;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  background: #420027b9;
  border: 5px solid #ccc;
  background-position: 50%;
  background-repeat: no-repeat;
  background-image: url('backCard.svg');
  -webkit-transform: rotateY(-180deg);
  -ms-transform: rotateY(-180deg);
  transform: rotateY(-180deg);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
      />
      <div
        class="c2 c4"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27120%27%20height=%27200%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="test"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>
</div>
`;
